cardano-api/src/Cardano/Api/Query.hs

```diff
@@ -166,17 +172,51 @@ newtype UTxO era = UTxO (Map TxIn (TxOut era))
 instance IsCardanoEra era => ToJSON (UTxO era) where
   toJSON (UTxO m) = Aeson.Array . Vector.fromList . map convert $ Map.toList m
    where
+    toText :: TxIn -> Text
+    toText txin = case toJSON txin of
+                    Aeson.String txt -> txt
+                    _ -> error "failure"
     convert :: (TxIn, TxOut era) -> Aeson.Value
-    convert ((TxIn txId (TxIx ix), txout)) =
-      let txin = ( Text.decodeUtf8 (serialiseToRawBytesHex txId)
-                 <> "#"
-                 <> Text.pack (show ix)
-                 )
-      in object [ txin .= toJSON txout]
+    convert (txin, txout) = object [ toText txin .= toJSON txout]
```

Comment:
You already have an ToJSONKey instance for TxIn, so you can just reuse that here. You don't need the toText function.

What you have here is an array of objects each with a single key. Why not make it a big JSON map/object using the txin as the keys?

I think this whole instance can be replaced by:
```
toJSON (UTxO m) = toJSON m
```
Because you've got `TxIn` as an instance of `ToJSONKey`, and you have a `ToJSON` instance for `TxOut`. So the existing general instance for of `ToJSON` for `Map` will just do the right thing I think.

Reply:
Thanks I missed this